From 7c29f1edca42f7e4216855d294a1a7d5b0bcb57f Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Tue, 29 Apr 2014 19:55:25 +0200
Subject: [PATCH 25/26] BUILD: config: remove a warning with clang

Commit fc6c032 ("MEDIUM: global: add support for CPU binding on Linux ("cpu-map")")
merged into 1.5-dev13 involves a useless test that clang reports as a warning. The
"low" variable cannot be negative here. Issue reported by Charles Carter.
---
 src/cfgparse.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cfgparse.c b/src/cfgparse.c
index 6ac3ccc..c4f092f 100644
--- a/src/cfgparse.c
+++ b/src/cfgparse.c
@@ -1405,7 +1405,7 @@ int cfg_parse_global(const char *file, int linenum, char **args, int kwm)
 					high = swap;
 				}
 
-				if (low < 0 || high >= sizeof(long) * 8) {
+				if (high >= sizeof(long) * 8) {
 					Alert("parsing [%s:%d]: %s supports CPU numbers from 0 to %d.\n",
 					      file, linenum, args[0], (int)(sizeof(long) * 8 - 1));
 					err_code |= ERR_ALERT | ERR_FATAL;
-- 
1.8.3.2

